packages/flutter/lib/src/widgets/image.dart

```diff
@@ -188,12 +188,6 @@ class _ImageState extends State<Image> {
   ImageInfo _imageInfo;
 
   @override
-  void initState() {
-    super.initState();
-    _resolveImage();
```

Comment:
Right, the existing code wasn't wrong since subsequent updates to MediaQuery and DefaultAssetBundle were handled by dependenciesChanged().  Moving this one time init to build() avoids the need to define some new way to indicate that an inherited reference widget at initState() time is safe. 

This appears to have been the only example of its kind in our code base. If more cases crop up I suppose we'll need to add a way to defeat the assertion added by this PR.